src/canvas.class.js

```diff
@@ -512,6 +512,14 @@
     _shouldClearSelection: function (e, target) {
       var activeObjects = this.getActiveObjects(),
           activeObject = this._activeObject;
+
+      var selectionKeyPressed = false;
+      this.selectionKeys.forEach(function (selectionKey) {
+        if (e[selectionKey]){
+          selectionKeyPressed = true;
+        }
+      });
```

Comment:
anyway this could be done with  this.selectionKeys.some(function(key) { return e[selectionKey]; }) ?  is still ES5 and i think has the same performance of forEach.